scrapypuppeteer/browser_managers/playwright_browser_manager.py

```diff
@@ -0,0 +1,368 @@
+            "back": self.go_back,
+            "forward": self.go_forward,
+            "scroll": self.scroll,
+            "screenshot": self.screenshot,
+            "action": self.action,
+            "recaptcha_solver": self.recaptcha_solver,
+            "har": self.har,
+            "fill_form": self.fill_form,
+        }
+
+    def process_request(self, request):
+        if isinstance(request, PuppeteerRequest):
+            endpoint = request.action.endpoint
+            action_function = self.action_map.get(endpoint)
+            if action_function:
+                return action_function(request)
+
+        if isinstance(request, CloseContextRequest):
+            return self.close_contexts(request)
+
+    def close_contexts(self, request: CloseContextRequest):
+        self.context_manager.close_contexts(request)
+
+    def close_used_contexts(self):
+        self.context_manager.close_browser()
+
+    def process_response(self, middleware, request, response, spider):
+        return response
+
+    def map_navigation_options(self, navigation_options):
+        if not navigation_options:
+            return {}
+        event_map = {
+            "load": "load",
+            "domcontentloaded": "domcontentloaded",
+            "networkidle0": "networkidle",
+            "networkidle2": "networkidle",
+        }
+        mapped_navigation_options = {}
+        if "timeout" in navigation_options:
+            mapped_navigation_options["timeout"] = navigation_options["timeout"]
+
+        waitUntil = navigation_options.get("waitUntil")
+
+        if waitUntil:
+            if isinstance(waitUntil, list):
+                event_hierarchy = [
+                    "load",
+                    "domcontentloaded",
+                    "networkidle2",
+                    "networkidle0",
+                ]
+                strictest_event = max(
+                    waitUntil, key=lambda event: event_hierarchy.index(event)
+                )
+            elif isinstance(waitUntil, str):
+                strictest_event = waitUntil
+
+            if strictest_event in event_map:
+                mapped_navigation_options["wait_until"] = event_map[strictest_event]
+
+        return mapped_navigation_options
+
+    def map_click_options(self, click_options):
+        if not click_options:
+            return {}
+        mapped_click_options = {
+            "delay": click_options.get("delay", 0.0),
+            "button": click_options.get("button", "left"),
+            "click_count": click_options.get("clickCount", 1),
+        }
+        return mapped_click_options
+
+    def map_screenshot_options(self, screenshot_options):
+        if not screenshot_options:
+            return {}
+        mapped_screenshot_options = {
+            "type": screenshot_options.get("type", "png"),
+            "quality": screenshot_options.get("quality", "png"),
+            "full_page": screenshot_options.get("fullPage", False),
```

Comment:
Well, `quality` must be an integer: https://playwright.dev/python/docs/api/class-page#page-screenshot-option-quality